drivers/i2c/i2c_imx.c

```diff
@@ -238,8 +238,10 @@ static int i2c_imx_transfer(const struct device *dev, struct i2c_msg *msgs,
 		if ((msgs->flags & I2C_MSG_RW_MASK) == I2C_MSG_READ) {
 			i2c_imx_read(dev, msgs->buf, msgs->len);
 		} else {
-			if (!i2c_imx_write(dev, msgs->buf, msgs->len)) {
-				goto finish; /* No ACK received */
+			if (msgs->len) {
```

Comment:
If you moved this check up to enclose the entire `/* Transfer data */` section that would also prevent breakage if somebody issues a read of length zero.

Reply:
Thanks for your review @pabigot !
True also read is broken with , i'll update my pr this weekend.